qiskit/algorithms/minimum_eigen_solvers/vqe.py

```diff
@@ -517,7 +532,7 @@ def compute_minimum_eigenvalue(
 
         if aux_operators is not None:
             aux_values = self._eval_aux_ops(opt_params, aux_operators, expectation=expectation)
```

Comment:
> Thanks for the reno - in the meanwhile it seems that another change is causing a conflict with vqe. If that can be sorted then this is good to go I think.

@woodsp-ibm The conflict is the following
```
<<<<<<< issue6772/dict_support_for_aux_ops
            aux_values = self._eval_aux_ops(opt_params, aux_operators, expectation=expectation)
            result.aux_operator_eigenvalues = aux_values
=======
            aux_values = self._eval_aux_ops(opt_result.x, aux_operators, expectation=expectation)
            result.aux_operator_eigenvalues = aux_values[0]
>>>>>>> main
```

where `opt_result` is obtained from the new `self.optimizer.minimize`.

I've locally changed the old `opt_params` to the new `opt_result.x` but this leads to nearly all VQE tests failing with NoneType errors. The culprit is that when `self.optimizer.minimize` encounters an error, it falls back to the deprecated `self.optimizer.optimize`. The results are stored in the `opt_result` variable but this is a tuple instead of an object with attributes (lines [505-511](https://github.com/Qiskit/qiskit-terra/pull/6870/files#diff-e058a38497350f6198cbb401e70fe7687e0aa0b7b3316f07ff751bc46fb4f6f9R505-R511)) so that `opt_result.x` returns None.

Any suggestions on what should be done next?

Reply:
Could you specify in a bit more detail what breaks? I just tried merging `main` into your branch locally and using the following:
```python
            aux_values = self._eval_aux_ops(opt_result.x, aux_operators, expectation=expectation)
            result.aux_operator_eigenvalues = aux_values
```
The unittests in `test/python/algorithms/test_vqe.py` pass just fine. I am not sure whether I am misunderstanding the error you were facing or whether this problem has been resolved in the past 6 days.

Reply:
That is good to hear, thank you for testing. It is probably an issue with my local repository then since the tests also fail when I'm simply on the main branch after updating it to the upstream main branch.

Just to be safe I'll merge via github instead of my local git repo.
Below is an excerpt of the errors I get from pytest.
```
test\python\algorithms\test_vqe.py FFFFFFFFFFFFF.FFFssssssssF      [100%]
_____ TestVQE.test_aux_operators_dict ______
'NoneType' object is not iterable

During handling of the above exception, another exception occurred:
NOTE: Incompatible Exception Representation, displaying natively:

testtools.testresult.real._StringException: Traceback (most recent call last):
  File "C:\QuantumComputing\qiskit-terra\qiskit\algorithms\minimum_eigen_solvers\vqe.py", line 508, in compute_minimum_eigenvalue
    opt_result = self.optimizer.minimize(
  File "C:\QuantumComputing\qiskit-terra\qiskit\algorithms\optimizers\scipy_optimizer.py", line 129, in minimize
    raw_result = minimize(

    (skipping part of the stack trace)

  File "C:\QuantumComputing\qiskit-terra\qiskit\dagcircuit\dagcircuit.py", line 1073, in substitute_node_with_dag
    node_map = self._multi_graph.substitute_node_with_subgraph(
AttributeError: 'PyDAG' object has no attribute 'substitute_node_with_subgraph'
```